src/jit/importer.cpp

```diff
@@ -3324,7 +3324,7 @@ GenTreePtr Compiler::impIntrinsic(GenTreePtr            newobjThis,
     //
     // We disable the inlining of instrinsics for MinOpts.
     //
-    if (!mustExpand && (opts.compDbgCode || opts.MinOpts()))
+    if (!mustExpand && (opts.compDbgCode || opts.MinOpts()) && !gtIsRecursiveCall(method))
```

Comment:
Can you please add a comment explaining the purpose of this extra check.

Also, while you're in the neighborhood, can you fix the misspelled word in the existing comment?

Reply:
Got it, will do.